packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItem.js

```diff
@@ -0,0 +1,119 @@
+import React, { useState, useRef, useEffect } from 'react';
+
+import { Popover, PopoverContent, Layer } from '@carbon/react';
+import PropTypes from 'prop-types';
+import { Add } from '@carbon/react/icons';
+
+import { ConditionBuilderButton } from '../ConditionBuilderButton/ConditionBuilderButton';
+import {
+  blockClass,
+  translateWithId,
+  valueRenderers,
+} from '../ConditionBuilderContext/DataConfigs';
+
+export const ConditionBuilderItem = ({
+  children,
+  className,
+  label,
+  renderIcon,
+  title,
+  isOpen,
+  type,
+  ...rest
+}) => {
+  const contentRef = useRef(null);
+
+  const [open, setOpen] = useState(false);
```

Comment:
this state is redundant. if we're tracking the open state in the parent we should continue to derive `open` from there instead of setting another local `open` state variable

Reply:
popover opening is handled with this local state only. But in some scenarios (add new condition, close single select dropdown) we need to rely on this open prop(renamed to popoverState for better understanding) as the default state. Component is re-rendered on every interactions.
Please see the [this link](https://deploy-preview-4951--carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-conditionbuilder--condition-builder) of the deployment of complete changes to see the latest changes. I have raised a [draft PR ](https://github.com/carbon-design-system/ibm-products/pull/4951) with all changes